analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -168,8 +171,7 @@ static void Traverse(ControlFlowBranch branch, HashSet<BasicBlock> result, IColl
         }
     }
 
-    private bool IsIgnoredLocation(SyntaxNode x) =>
-        IsForLoopIncrementor(x)
-        || IsConditionalAccessExpression(x)
-        || IsLeftCoalesceExpression(x);
+    // For SwitchExpressionArms like `true => 5` we are only interested in the left part (`true`).
+    private static SyntaxNode ToBranchValueCondition(SyntaxNode syntax) =>
+        syntax.IsKind(SyntaxKindEx.SwitchExpressionArm) ? ((SwitchExpressionArmSyntaxWrapper)syntax).Pattern : syntax;
```

Comment:
This gets the switch expression condition otherwise it returns the syntax directly - correct?

I would delete this method and inline this (with the comment) directly in line 94.
IMO it's clearer. I also find the name a bit confusing.